packages/typo3-docs-theme/src/EventListeners/AddThemeSettingsToProjectNode.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\Typo3DocsTheme\EventListeners;
+
+use phpDocumentor\Guides\Event\PostProjectNodeCreated;
+use phpDocumentor\Guides\Nodes\Inline\PlainTextInlineNode;
+use T3Docs\Typo3DocsTheme\Settings\Typo3DocsThemeSettings;
+
+final class AddThemeSettingsToProjectNode
+{
+    public function __construct(
+        private readonly Typo3DocsThemeSettings $themeSettings,
+    ) {}
+
+    public function __invoke(PostProjectNodeCreated $event): void
+    {
+        $projectNode = $event->getProjectNode();
+        foreach ($this->themeSettings->getAllSettings() as $key => $setting) {
+            if ($setting !== '') {
+                $projectNode->addVariable($key, new PlainTextInlineNode(strval($setting)));
```

Comment:
You check for an non-empty string above, but then cast it to a string with strval(). Either the check should be adjusted (maybe null?) or the strval is superfluous.

Reply:
true, even though the $settings where supposed to be `array<string, string>` still a boolean value was passed. (wondering why phpstan didnt catch this). Anyway, I fix it now